packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -49,19 +53,50 @@ const VirtualizedComboBoxMenu = (props: React.ComponentProps<'div'>) => {
     );
   }, [focusedIndex, menuRef]);
 
-  const { outerProps, innerProps, visibleChildren } = useVirtualization({
+  const virtualizer = useVirtualScroll({
     // 'Fool' VirtualScroll by passing length 1
     // whenever there is no elements, to show empty state message
-    itemsLength: filteredOptions.length || 1,
-    itemRenderer: virtualItemRenderer,
+    count: filteredOptions.length || 1,
+    getScrollElement: () => menuRef.current,
+    estimateSize: () => (mostlySubLabeled ? 48 : 36),
     scrollToIndex: focusedVisibleIndex,
+    gap: -1,
   });
 
+  const virtualItemRenderer = React.useCallback(
+    (virtualItem: VirtualItem) => {
+      const menuItem =
+        filteredOptions.length > 0
+          ? getMenuItem(filteredOptions[virtualItem.index], virtualItem.index)
+          : (children as JSX.Element); // Here is expected empty state content
+      return React.cloneElement(menuItem, {
+        key: virtualItem.key,
+        ref: virtualizer.measureElement,
+        style: {
+          position: 'absolute',
+          top: 0,
+          left: 0,
+          width: '100%',
+          transform: `translateY(${virtualItem.start}px)`,
+        },
+      });
+    },
+    [filteredOptions, getMenuItem, children, virtualizer.measureElement],
+  );
+
   return (
-    <Box as='div' {...outerProps} {...rest}>
-      <div {...innerProps} ref={innerProps.ref}>
-        {visibleChildren}
-      </div>
+    <Box
```

Comment:
Since we are now exploring the idea of hiding this `<div>` inside shadow DOM in `Table` (https://github.com/iTwin/iTwinUI/pull/2092#discussion_r1644627752) and also in `Tree` (https://github.com/iTwin/iTwinUI/pull/2092#discussion_r1649220865), maybe it makes sense to also do the same in `ComboBox`? It would keep the light DOM consistent regardless of whether virtual scroll is enabled.

/cc @r100-stack

Reply:
Yeah, I think that makes sense. Wrapped that `<div>` in `ShadowRoot` in this [commit](https://github.com/iTwin/iTwinUI/pull/2061/commits/7b20c9d63b680634cd7e56ddd77482644eb4cfb5).